### PR TITLE
Relax Segment ActivityTypes

### PR DESCRIPTION
**Pull request by JohnScolaro**
closes https://github.com/stravalib/stravalib/issues/432

_The text above will ensure the issue will be closed when this PR is merged_

## Description

See https://github.com/stravalib/stravalib/issues/432 for a detailed explanation of the issue. This is a fix I made that works, but I don't know if it's the correct way of doing it, so I'm putting it here for review. Essentially when downloading activities (that aren't rides and runs) that have segments, Pydantic fails to validate the activities because it's expecting the `ActivityType` in the `Segment` in the `SegmentEffort` in the `Activity` to be either ride or run. Since Strava has been adding segments for new activity types, this breaks.

## Type of change

Select the statement best describes this pull request.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [ ] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [x] No, i'd like some help with tests
- [ ] This change doesn't require tests

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

_If you are having a hard time getting the tests to run correctly feel free to
ping one of the maintainers here!_

### CODE REVIEW

**Comment by lwasser** [comment]
thank you for this pr @JohnScolaro !! i just fixed a merge conflict and gave the pr access to run Ci! i understand generally what the good of this is about - we had just added those typed literal values for run and ride but it seems like that then causes some issues with interfacing with other activity types (which presumably aren't documented in the API?)

File: src/stravalib/model.py [hunk 1]
```diff
@@ -938,7 +938,7 @@ class Segment(
     map: Optional[Map] = None
     athlete_segment_stats: Optional[AthleteSegmentStats] = None
     athlete_pr_effort: Optional[AthletePrEffort] = None
-    activity_type: Optional[Literal["Ride", "Run"]] = None
+    activity_type: Optional[ActivityType] = None
```
Comment (lwasser) [inline src/stravalib/model.py:941]: @jsamoocha i'm pretty sure it's ok for us to relax this - as it's pretty much the way things were previously. but i also remember another part of this module where i had to add a literal type for one of the methods. i think @enadeau helped me with that. so i just want to make sure this fix will actually work

Comment (lwasser) [reply src/stravalib/model.py:941]: oh perhaps that was athlete type?

Comment (jsamoocha) [reply src/stravalib/model.py:941]: This same fix was applied in #369. It led to typing errors (which we see now in CI). I have no idea ATM if we have to choose between two "evils" or if there's a (typing) workaround.

**Comment by enadeau** [comment]
I don't see any more elegant solution at the moment. Of course it would be nice to see Strava fix the open api specification, but there's not much we can do about that. I think this is the best fix in the meantime.

I've added a test to make sure we don't regress this issue again and fixed the last mypy issue.

If another maintainer approves the change, I'll merge it and include that in the 1.5 release.

File: src/stravalib/model.py (hunk 1 above)
Comment (lwasser) [reply src/stravalib/model.py]: ahhh ok thanks. so we should merge following @enadeau commits below.

**Review by lwasser — APPROVED** [review]
ok so we are back to the original work around with a test to make sure this works as expected. thank you @JohnScolaro and @enadeau for this. i'm approving it.

**Comment by JohnScolaro** [comment]
Thanks guys :)

**Comment by enadeau** [comment]
Merged :rocket:  Thanks @JohnScolaro for your contribution

**Pull request merged**